src/core/RegistryAdapter.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "../interfaces/IERC7484.sol";
+
+abstract contract RegistryAdapter {
+    event ERC7484RegistryConfigured(address indexed smartAccount, address indexed registry);
+
+    IERC7484 internal $registry;
+
+    modifier withRegistry(address module, uint256 moduleTypeId) {
+        IERC7484 registry = $registry;
+        if (address(registry) != address(0)) {
+            registry.check(module, moduleTypeId);
+        }
+        _;
+    }
+
+    function setRegistry(
```

Comment:
Where is the latest code to the registry? (repo link)
Is it audited?
do you guys have deployed registry on some networks?

Reply:
https://github.com/rhinestonewtf/registry/

Audit is done but in remediation atm

Reply:
also probably good to rename it to setModuleRegistry and have ERC7484 mention in the natspecs